datahub-graphql-core/src/main/resources/gms.graphql

```diff
@@ -265,7 +269,12 @@ type Dataset implements EntityWithRelationships & Entity {
     """
     Schema metadata of the dataset
     """
-    schema: Schema
+    schema: Schema @deprecated(reason: "Use `schemaMetadata`")
+
+    """
+    Schema metadata of the dataset
+    """
+    schemaMetadata(version: Long): SchemaMetadata
```

Comment:
ty for making this consistent. i should have done so in the first place.

Reply:
btw. what if we want to know the available versions? do we need to support that eventually ?

Reply:
We can discuss this later- right now if you fetch `-1` you get all available versions by checking the `version` of the result we get back from fetch `-1`. When we need a more explicit way to fetch this value we can look into it then.

Reply:
very scary. caller need to know about -1 stuff

Reply:
Sure, as more use cases arise, lets use those to understand the needs of fetching all version count. We can have another method to do this, or include max version in every versioned aspect.